### PR TITLE
global: migrate CI to GH Actions

### LINKED ISSUE #219: global: migration to github-actions from travis

### PR DESCRIPTION
**Pull request by ParthS007**
closes #219

### CODE REVIEW

**Comment by ParthS007** [comment]
Build failing because of https://github.com/inveniosoftware/invenio-oauthclient/issues/195

File: run-tests.sh [hunk 1]
```diff
@@ -6,10 +6,15 @@
...
-isort -rc -c -df **/*.py && \
-check-manifest --ignore ".travis-*" && \
-sphinx-build -qnNW docs docs/_build/html && \
-python setup.py test && \
+# Quit on unbound symbols
+set -o nounset
+
+pydocstyle invenio_oauth2server
+isort -rc -c -df **/*.py
+check-manifest --ignore ".travis-*"
```
Comment (lnielsen) [inline run-tests.sh:17]:
```suggestion
check-manifest --ignore ".*-requirements.txt"
```

**Comment by diegodelemos** [comment]
Minor: simplify [test requirements](https://github.com/inveniosoftware/invenio-oauth2server/blob/02012fb35b07491b69e9b9031ee6f9ff8b10d1db/setup.py#L19-L30) through `pytest-invenio` like [here](https://github.com/inveniosoftware/invenio-theme/pull/215/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7L19).

**Comment by diegodelemos** [comment]
Minor: `pytest.ini` to be updated according to [cookiecutter-invenio-module's `pytest.ini`](https://github.com/inveniosoftware/cookiecutter-invenio-module/blob/master/%7B%7B%20cookiecutter.project_shortname%20%7D%7D/pytest.ini).

**Comment by diegodelemos** [comment]
Minor: replace all occurences of `.travis.yml` (e.g. https://github.com/inveniosoftware/invenio-oauth2server/blob/02012fb35b07491b69e9b9031ee6f9ff8b10d1db/.editorconfig#L38)

File: run-tests.sh [hunk 2]
```diff
@@ -6,10 +6,15 @@
...
+set -o errexit
 
-pydocstyle invenio_oauth2server && \
-isort -rc -c -df **/*.py && \
-check-manifest --ignore ".travis-*" && \
-sphinx-build -qnNW docs docs/_build/html && \
-python setup.py test && \
+# Quit on unbound symbols
+set -o nounset
+
```
Comment (diegodelemos) [inline run-tests.sh:14]: Major: ⚠️ We are setting `postgreqlX` and `mysqlX` in the build matrix but we never call `docker-services-cli` here, so basically test won't be able to connect to a DB that isn't running.

File: run-tests.sh [hunk 3]
```diff
@@ -5,11 +5,19 @@
...
+pydocstyle invenio_oauth2server
+isort -rc -c -df **/*.py
+check-manifest --ignore ".travis-*"
+sphinx-build -qnNW docs docs/_build/html
+docker-services-cli up ${DB}
+python setup.py test
 sphinx-build -qnNW -b doctest docs docs/_build/doctest
```
Comment (diegodelemos) [inline run-tests.sh:23]: Moderate: Missing the `docker-services-cli down` see https://github.com/inveniosoftware/cookiecutter-invenio-module/blob/e3ca55f18a3e35ed4d3f5875d9647d408c45295b/%7B%7B%20cookiecutter.project_shortname%20%7D%7D/run-tests.sh#L12-L14

**Review by diegodelemos — APPROVED** [review]

**Pull request merged**